test/e2e/framework.go

```diff
@@ -1177,6 +1149,95 @@ func (data *TestData) deployFlowAggregator() error {
 	return nil
 }
 
+// getSparkOperator retrieves the name of the Spark Operator Pod (policy-reco-spark-operator-*).
+func (data *TestData) getSparkOperator() (string, error) {
+	listOptions := metav1.ListOptions{
+		LabelSelector: "app.kubernetes.io/name=spark-operator",
+	}
+	var pod *corev1.Pod
+	if err := wait.Poll(defaultInterval, defaultTimeout, func() (bool, error) {
```

Comment:
It's not entirely clear to me how wait.Poll works. If the timeout expires would it return an error?

If not, maybe we should return an error if pod.Name is nil, otherwise we're good.

Reply:
Yes, if it hits a timeout, it will return an ErrWaitTimeout error in L1169.